app/service/CasesService.scala

```diff
@@ -20,14 +20,27 @@ import connectors.BindingTariffClassificationConnector
 import javax.inject.{Inject, Singleton}
 import models.{Case, NewCaseRequest}
 import uk.gov.hmrc.http.HeaderCarrier
+import scala.concurrent.ExecutionContext.Implicits.global
 
 import scala.concurrent.Future
 
 @Singleton
-class CasesService @Inject()(connector: BindingTariffClassificationConnector){
+class CasesService @Inject()(connector: BindingTariffClassificationConnector) {
 
   def create(c: NewCaseRequest)(implicit hc: HeaderCarrier): Future[Case] = {
     connector.createCase(c)
   }
 
+  def getCaseForUser(userEori: String, reference: String)(implicit hc: HeaderCarrier): Future[Option[Case]] = {
+
+    connector.findCase(reference).map {
+      case Some(c) if caseBelongsToUser(c, userEori) => Some(c)
+      case _ => None
+    }
+  }
+
+  private def caseBelongsToUser(c: Case, eoriNumber: String): Boolean = {
```

Comment:
this code is not very readable
Also the condition `c.application.holder.eori == eoriNumber` is written twice

Reply:
Just a suggestion but we could add a method to ‘case’ called ‘hasEORI(eori: String):Boolean’

Reply:
Done

Reply:
did you commit and push?
I cannot see the change

Reply:
Can you extract `c.application.holder.eori == eoriNumber` to a boolean method?

Reply:
Yes - forgot to push last night